### DISCUSSION TITLE
What's the equivalent of webpack `externals` when using `--turbo` with a nextjs app?

### DISCUSSION THREAD

**Discussion by gpxl-dev** [vercel/next.js#69833] [Help]
### Summary

I need the equivalent of this for turbopack:

```
const nextConfig = {
  webpack: (config) => {
    config.externals.push(
      "pino-pretty",
      "encoding"
    );
    return config;
  },
}
```

But can't find the option anywhere... Is it possible?

**Comment by shifenhutu** [comment]
@gpxl-dev   friend , did you fix it ?

**Comment by mischnic** [comment]
Use `serverExternalPackages`: https://nextjs.org/docs/app/api-reference/next-config-js/serverExternalPackages
```js
const nextConfig = {
  serverExternalPackages: ['pino-pretty', 'encoding'],
}
 
module.exports = nextConfig
```

**Reply by songkeys** [reply]
After having done this way, I still got the `Module not found: Can't resolve 'pino-pretty'` when building.

<img width="657" alt="SCR-20241023-tllo" src="https://github.com/user-attachments/assets/eed99589-972f-42bc-8158-4f74b6d2a68c">

edit: found the issue: https://github.com/vercel/next.js/issues/68805

**Reply by AbdurRahamanAR** [reply]
@mischnic Does't that only for server component or Route.js (Telling from docs)?
